main.go

```diff
@@ -64,117 +80,267 @@ Usage:
 
 	// deadLinks provides a lists to contain all failed url links that are
 	// considered deadlinks.
-	var deadLinks []*linkReport
+	var deadLinks []linkReport
+
+	start := time.Now().UTC()
+
+	dead := make(chan *linkReport)
 
-	dead := collectFrom(path, !*hostOnly)
+	go collectFrom(path, !*hostOnly, *workers, dead)
 
 	for {
 		select {
+		case <-time.After(time.Second):
+			continue
+
 		case link, ok := <-dead:
 			if !ok {
-				fmt.Println("--------------------DEAD LINKS------------------------------")
+				end := time.Now().UTC()
+				fmt.Println("--------------------Timelapse-------------------------------")
+				fmt.Printf(`
+Start Time: %s
+End Time: %s
+Duration: %s
+`, start, end, end.Sub(start))
 
 				if len(deadLinks) > 0 {
+					fmt.Println("--------------------DEAD LINKS------------------------------")
 					for _, f := range deadLinks {
 						fmt.Printf(`
-			URL: %s
-			Status Code: %d
+URL: %s
+Status Code: %d
 
-			`, f.Link, f.Status)
+`, f.Link, f.Status)
 					}
-				} else {
-					fmt.Println("No Dead Links Found! Woot!")
+					os.Exit(-1)
+					return
 				}
 
 				fmt.Println("------------------------------------------------------------")
 
-				os.Exit(-1)
+				os.Exit(0)
```

Comment:
You don't need to explicitly issue os.Exit(0). Just by returning from the main function it will exit with a status code of 0.